### PR TITLE
JDK 14 testing

### PR DESCRIPTION
**Pull request by yschimke**
PR Branch for testing against pre-release builds of JDK 14.  Not to land before GA.

2019/12/12 |   | Rampdown Phase One (fork from main line)
-- | -- | --
2020/01/16 |   | Rampdown Phase Two
2020/02/06 |   | Initial Release Candidate
2020/02/20 |   | Final Release Candidate
2020/03/17 |   | General Availability

### CODE REVIEW

**Comment by yschimke** [comment]
https://jdk.java.net/14/release-notes

Weak Named Curves in TLS, CertPath, and Signed JAR Disabled by Default (JDK-8233228)
Removed SSLv2Hello and SSLv3 From Default Enabled TLS Protocols (JDK-8190492)
Added 4 Amazon Root CA Certificates (JDK-8233223)
Added LuxTrust Global Root 2 Certificate (JDK-8232019)
SunJCE Provider Throws NoSuchAlgorithmException for AES/GCM/PKCS5Padding (JDK-8180392)
Deprecated the Legacy Elliptic Curves for Removal (JDK-8234924)

**Comment by yschimke** [comment]
Failing in gradle

```
java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7
	at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)
	at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)
	at org.codehaus.groovy.reflection.ClassInfo.<clinit>(ClassInfo.java:107)
	at org.codehaus.groovy.reflection.ReflectionCache.getCachedClass(ReflectionCache.java:95)
	at org.codehaus.groovy.reflection.ReflectionCache.<clinit>(ReflectionCache.java:39)
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.registerMethods(MetaClassRegistryImpl.java:210)
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:108)
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:86)
	at groovy.lang.GroovySystem.<clinit>(GroovySystem.java:36)
```

**Comment by yschimke** [comment]
https://github.com/gradle/gradle/issues/10248 

> Unless something prevents us from doing it, the plan is to have JDK14 support in 6.3 (due in a couple of weeks).

File: okhttp/src/test/java/okhttp3/internal/connection/RouteSelectorTest.java [hunk 1]
```diff
@@ -467,7 +470,10 @@
   @Test public void routeToString() throws Exception {
     Route route = new Route(httpAddress(), Proxy.NO_PROXY,
         InetSocketAddress.createUnresolved("host", 1234));
-    assertThat(route.toString()).isEqualTo("Route{host:1234}");
+    assertThat(route.toString()).isEqualTo(
+        PlatformVersion.INSTANCE.getMajorVersion() >= 14
+            ? "Route{host/<unresolved>:1234}"
```
Comment (dave-r12) [inline okhttp/src/test/java/okhttp3/internal/connection/RouteSelectorTest.java]: Interesting!

**Pull request merged**